### DISCUSSION TITLE
How to install searx in windows?

### DISCUSSION THREAD

**Discussion by vigneshchalapathy** [searx/searx#821] [Installation issue: custom]
#Can anyone suggest me a way to install searx  which permits me to customize it to the core.

I use apache from xampp(windows 10 operating system)

I followed these installation procedures for apache.

Add wsgi mod:

sudo apt-get install libapache2-mod-uwsgi
sudo a2enmod uwsgi

Add this configuration in the file /etc/apache2/apache2.conf:

Snippet:

<Location />
    Options FollowSymLinks Indexes
    SetHandler uwsgi-handler
    uWSGISocket /run/uwsgi/app/searx/socket
</Location>

Note that if your instance of searx is not at the root, you should change <Location /> by the location of your instance, like <Location /searx>.

Restart Apache:
sudo /etc/init.d/apache2 restart

But when  i attempt to paste this snippet in  httpd.conf file,

Apache fails to run.

Also suggest me anyother basic installation procedures for windows.

**Comment by dalf** [comment]
If you use xampp, you should install mod_wsgi.

I think, but can't confirm that ```sudo apt-get install libapache2-mod-uwsgi``` installs mod_wsgi for apache, the one included in embedded Linux (The Ubuntu included in Windows).

So either : 
* you choose to use xampp, and you have to adapt the installation steps for Windows (python for windows, wsgi for windows, etc...). See https://code.google.com/archive/p/modwsgi/wikis/InstallationOnWindows.wiki and http://enkoding.blogspot.fr/2013/01/setup-python-for-web-in-7-steps-on.html
* or you use the embedded Linux. Install apache using ```apt install apache2```. The point I don't know is how to turn apache2 into a service. You can still start it by hand using ```service start apache2```

**Comment by vigneshchalapathy** [comment]
Thano you,I will check

**Comment by vigneshchalapathy** [comment]
Can  searx  be installed with Django or http://www.ampps.com/downloads server

**Comment by dalf** [comment]
It's a different topic but : 
* there is no specific integration with Django. Searx just uses the HTTP protocol to produce HTML, JSON, RSS, CSV output.
* ampps uses Python 3.5, for now searx is written in python 2. There a port python 3 under development in the [py3 branch](https://github.com/asciimoo/searx/tree/py3/searx). mod_wsgi is included according to the documentation, so it should works as long as you can install the searx dependencies.
